Source/ProjectRimFactory/Storage/UI/ITab_Items.cs

```diff
@@ -47,6 +50,28 @@ public ITab_Items()
 
         protected Building_StorageUnitIOPort SelectedIOPort => SelThing as Building_StorageUnitIOPort;
 
+        private static Dictionary<Thing, List<Pawn>> canBeConsumedby = new Dictionary<Thing, List<Pawn>>();

```

Comment:
Any particular reason `static`?

They run the risk of getting really big (and containing references to destroyed items, etc) if the game runs for a long time, if the player restarts the game, etc.

If you make it private, it gets created once for the ITab; I don't know what impact that would have on first opening the ITab, but it might be small enough that private is the way to go?

Reply:
it needs to be static so that i can access it from ` public static List<FloatMenuOption> ChoicesForThing(Thing thing, Pawn pawn, string thingLabelShort)`

removing the static there results in immense null ref issues

but we should definitively clear them at some point just to be sure.
maybe if their size is X times larger that the max size (items) from the DSU?

Reply:
Here's a way that might work:

```CSharp
public static ITab_Items currentOpenITab;
...
public Dictionary<Thing, List<Pawn>> canBeConsumedBy = ...
...
OnOpen() {
  currentOpenITab=this;
}
```

Yes?